api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java

```diff
@@ -150,12 +153,25 @@ public DiagnosticReport saveFHIRDiagnosticReport(DiagnosticReport diagnosticRepo
 	}
 
 	@Override
-	public DiagnosticReport purgeFHIRDiagnosticReport(DiagnosticReport diagnosticReport) {
-		// Delete `Name` Obs
-		// Delete `Status` Obs
-
-		// Delete Obs (`Result` as Set of Obs)
+	public void retireFHIRDiagnosticReport(String id) {
+		EncounterService encounterService = Context.getEncounterService();
 		// Delete Binary Obs Handler which used to store `PresentedForm`
-		return diagnosticReport;
+
+		// Delete Encounter OpenMRS Object
+		Encounter omrsDiagnosticReport = encounterService.getEncounterByUuid(id);
```

Comment:
So we are only need to retire the encounter? Is there any additional resources need to retire as well? May be Obs associate with the encounter retired when encounter retires by underline API?